File: cses/leetcode/cpp/2373.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<vector<int>> largestLocal(vector<vector<int>> &grid)
    {
        int center_x = 1, center_y = 1;
        int rows = grid.size(), cols = grid[0].size();
        vector<vector<int>> answer;
        for (; center_x <= rows - 2; center_x++)
        {
            vector<int> temp;
            center_y = 1;
            for (; center_y <= cols - 2; center_y++)
            {
                int max_value = max({
                    grid[center_x - 1][center_y - 1],
                    grid[center_x - 1][center_y],
                    grid[center_x - 1][center_y + 1],
                    grid[center_x][center_y - 1],
                    grid[center_x][center_y],
                    grid[center_x][center_y + 1],
                    grid[center_x + 1][center_y - 1],
                    grid[center_x + 1][center_y],
                    grid[center_x + 1][center_y + 1],
                });
                temp.push_back(max_value);
            }
            answer.push_back(temp);
        }
        return answer;
    }
};

File: cses/leetcode/javascript/2373.js
/**
 * @param {number[][]} grid
 * @return {number[][]}
 */
const largestLocal = function(grid) {
    let rows = grid.length;
    let cols = grid[0].length;
    let origin_x = 1, origin_y = 1;
    let ans = [];
    let counter = 0;
    for (; origin_x <= rows - 2; origin_x++) {
        let temp = [];
        let i = 0;
        origin_y = 0; // Reset origin_y for each row
        for (; origin_y <= cols - 2; origin_y++) {
            let value = Math.max(
                grid[origin_x - 1][origin_y - 1],
                grid[origin_x - 1][origin_y],
                grid[origin_x - 1][origin_y + 1],
                grid[origin_x][origin_y - 1],
                grid[origin_x][origin_y],
                grid[origin_x][origin_y + 1],
                grid[origin_x + 1][origin_y - 1],
                grid[origin_x + 1][origin_y],
                grid[origin_x + 1][origin_y + 1]
            ); 
            temp[i] = value;
            i++;
        }
        ans.push(temp);
    }
    return ans;
};

var matrixArray = ()=>{
    const a = new Array(4);
    for (let i = 0; i < a.length; i++) {
       a[i] = new Array(4);
       for (let j = 0; j < a[i].length; j++) {
        a[i][j] = `[${i} ${j}]`;
       }
    }
}
matrixArray();